sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.cs

```diff
@@ -68,5 +74,62 @@ private static bool ShouldIncludeAutoGeneratedAttributes(GeneratorExecutionConte
 
             return string.Equals(value, bool.TrueString, System.StringComparison.OrdinalIgnoreCase);
         }
+
+        private IList<IMethodSymbol> GetEntryAssemblyFunctions(List<MethodDeclarationSyntax> candidateMethods, GeneratorExecutionContext context)
+        {
+            IList<IMethodSymbol>? entryAssemblyFuncs = new List<IMethodSymbol>();
+
+            foreach (MethodDeclarationSyntax method in candidateMethods)
+            {
+                var model = context.Compilation.GetSemanticModel(method.SyntaxTree);
+
+                if (FunctionsUtil.IsValidFunctionMethod(context, context.Compilation, model,  method))
+                {
+                    IMethodSymbol? methodSymbol = (IMethodSymbol) model.GetDeclaredSymbol(method)!;
+                    entryAssemblyFuncs.Add(methodSymbol);
+                }
+            }
+
+            return entryAssemblyFuncs.Count > 0 ? entryAssemblyFuncs : ImmutableList<IMethodSymbol>.Empty;
+        }
+
+        /// <summary>
+        /// Collect methods with Function attributes on them from dependent/referenced assemblies.
+        /// </summary>
+        private IList<IMethodSymbol> GetDependentAssemblyFunctions(GeneratorExecutionContext context)
```

Comment:
nit, use `IEnumerable<IMethodSymbol>` and `yield return` - it will be more efficient as you already `.ToList()` the results.